From 91ba22d2ee9514d21147183058c40d5038164975 Mon Sep 17 00:00:00 2001
From: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Date: Tue, 16 Aug 2022 15:50:14 +0200
Subject: [PATCH 203/233] [FIO tonxp] board: verdin-imx8mm: add USB power hooks

This is required to power the USB.

Upstream-Status: Inappropriate [other]
- NXP downstream specific

Signed-off-by: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Signed-off-by: Andrejs Cainikovs <andrejs.cainikovs@toradex.com>
(cherry picked from commit 0b3c9cd1ee406439afb14390681c653db6c09651)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 board/toradex/verdin-imx8mm/verdin-imx8mm.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/board/toradex/verdin-imx8mm/verdin-imx8mm.c b/board/toradex/verdin-imx8mm/verdin-imx8mm.c
index 55c02653da6..2b4dadd469d 100644
--- a/board/toradex/verdin-imx8mm/verdin-imx8mm.c
+++ b/board/toradex/verdin-imx8mm/verdin-imx8mm.c
@@ -14,6 +14,7 @@
 #include <micrel.h>
 #include <miiphy.h>
 #include <netdev.h>
+#include <usb.h>
 
 #include "../common/tdx-cfg-block.h"
 
@@ -52,6 +53,24 @@ int board_mmc_get_env_dev(int devno)
 	return devno;
 }
 
+int board_usb_init(int index, enum usb_init_type init)
+{
+	debug("%s: %d, type %d\n", __func__, index, init);
+
+	imx8m_usb_power(index, true);
+
+	return 0;
+}
+
+int board_usb_cleanup(int index, enum usb_init_type init)
+{
+	debug("%s: %d, type %d\n", __func__, index, init);
+
+	imx8m_usb_power(index, false);
+
+	return 0;
+}
+
 static enum pcb_rev_t get_pcb_revision(void)
 {
 	struct udevice *bus;
-- 
2.25.1

